benchmarks/Microsoft.AspNetCore.Routing.Performance/Matching/JumpTableSingleEntryBenchmark.cs

```diff
@@ -67,12 +86,234 @@ public int Implementation()
             var segments = _segments;
 
             var destination = 0;
-            for (var i = 0; i < strings.Length; i++)
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _implementation.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int Prototype()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _prototype.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int Trie()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
             {
-                destination = _table.GetDestination(strings[i], segments[i]);
+                destination = _trie.GetDestination(strings[i], segments[i]);
             }
 
             return destination;
         }
+
+        [Benchmark(OperationsPerInvoke = 5)]
+        public int VectorTrie()
+        {
+            var strings = _strings;
+            var segments = _segments;
+
+            var destination = 0;
+            for (int i = 0; i < strings.Length; i++)
+            {
+                destination = _vectorTrie.GetDestination(strings[i], segments[i]);
+            }
+
+            return destination;
+        }
+
+        private class SingleEntryAsciiVectorizedJumpTable : JumpTable
+        {
+            private readonly int _defaultDestination;
+            private readonly int _exitDestination;
+            private readonly string _text;
+            private readonly int _destination;
+
+            private readonly ulong[] _values;
+            private readonly int _residue0Lower;
+            private readonly int _residue0Upper;
+            private readonly int _residue1Lower;
+            private readonly int _residue1Upper;
+            private readonly int _residue2Lower;
+            private readonly int _residue2Upper;
+
+            public SingleEntryAsciiVectorizedJumpTable(
+                int defaultDestination,
+                int exitDestination,
+                string text,
+                int destination)
+            {
+                _defaultDestination = defaultDestination;
+                _exitDestination = exitDestination;
+                _text = text;
+                _destination = destination;
+
+                int length = text.Length;
+                ReadOnlySpan<char> span = text.ToLowerInvariant().AsSpan();
```

Comment:
var

Reply:
var all over the class

Reply:
YEAH VS did this. I had to fix my settings....